AudioKit/Common/MIDI/AKMIDI+Sending.swift

```diff
@@ -230,24 +223,11 @@ extension AKMIDI {
     ///
     public func closeOutput(uid outputUid: MIDIUniqueID) {
         let name = destinationName(for: outputUid)
-        AKLog("Closing MIDI Input '\(String(describing: inputName))'")
+        AKLog("Closing MIDI Output '\(String(describing: name))'")
         var result = noErr
         if let endpoint = endpoints[outputUid] {
-            result = MIDIPortDisconnectSource(outputPort, endpoint)
-            if result == noErr {
                 endpoints.removeValue(forKey: outputUid)
-
-                result = MIDIPortDispose(outputPort)
```

Comment:
Maybe this function still passes? Can you try this keeping this line intact and just skip the MIDIPortDisconnectSource?

Reply:
This function passes but it seems to close all midi outputs altogether. I.e. if I have the following situation:
- Output 1 opened
- Output 2 opened
and then close Output 1, midi will not be sent to Output 2 anymore because of this (or any other output if you have more). 

From what I experimented, it seems that the outputPort variable functions as a sort of global midi output port. It just has to be opened once for the first midi output and then when you attempt to open subsequent midi outputs it suffices to add them to the endpoints listing and midi will be sent to them.

Therefore, when you open a new midi output, I've seen that you don't necessarily have to assign a new value to outputPort via tempPort = MIDIOutputPort(client: client, name: outputPortName), unless of course the outputPort hasn't been initialized yet.

Referring strictly to the MIDIPortDispose part you talked about above, I would use it only when there are no more endpoints left in the endpoints listing, i.e. when you have closed all midi output ports.

Reply:
That's correct @andreiantonescu.  I find it often helpful to refer to this diagram I drew of CoreMIDI:

https://github.com/AudioKit/AudioKit/blob/master/docs/CoreMIDI%20Design.svg

You also should be careful to not dispose of physical ports that you don't create.

Reply:
Thanks for the info @karnlund, I've pushed a new commit to address this.